src/tiler.coffee

```diff
@@ -95,121 +96,91 @@
+        'reverse'
+      else
+        ''
 
-      # Determine the direction of animation
-      #
-      if @_isNavigatingForward(enterTileId) || !@options.reverseSupport || !customReverse == true
-        exitTileInitialState      = 'exit'
-        exitTileInitialPosition   = 'start'
-        exitTileFinalPosition     = 'end'
+      # Disable animations
+      @element.addClass 'animation-disabled'
 
-        enterTileInitialState     = 'enter'
-        enterTileInitialPosition  = 'start'
-        enterTileFinalPosition    = 'end active'
+      # Build tile starting position animations classes
+      enterStartClass = "tiler-tile #{animationClass} active enter #{reverseClass} start"
+      exitStartClass = "tiler-tile #{animationClass} previous exit #{reverseClass} start"
+      otherTileClass = 'tiler-tile'
 
-      else
-        exitTileInitialState      = 'enter'
-        exitTileInitialPosition   = 'end'
-        exitTileFinalPosition     = 'start'
-
-        enterTileInitialState     = 'exit'
-        enterTileInitialPosition  = 'end'
-        enterTileFinalPosition    = 'start active'
-
-      # Setup tiles without animations
-      @$exitTile.add(@$enterTile).css
-        'transition-duration': '0 !important'
-        '-o-transition-duration': '0 !important'
-        '-moz-transition-duration': '0 !important'
-        '-webkit-transition-duration': '0 !important'
-
-      # Add start state classes
-      @$exitTile.attr 'class', "tiler-tile #{exitTileInitialState} #{exitTileInitialPosition} #{animationClass}"
-      @$enterTile.attr 'class', "tiler-tile #{enterTileInitialState} #{enterTileInitialPosition} #{animationClass}"
-
-      # Restore animation duration
-      @$exitTile.add(@$enterTile).css
-        'transition-duration': ''
-        '-o-transition-duration': ''
-        '-moz-transition-duration': ''
-        '-webkit-transition-duration': ''
-
-      # Swap classes to animate
-      @$exitTile.switchClass exitTileInitialPosition, exitTileFinalPosition
-      @$enterTile.switchClass enterTileInitialPosition, enterTileFinalPosition
+      # Set tile classes
+      @$enterTile.attr 'class', enterStartClass
+      @$exitTile.attr 'class', exitStartClass
+      @$tiles.not(@$enterTile).not(@$exitTile).attr 'class', otherTileClass
+
+      # setTimeout needed to give the browser time to repaint the tiles (if neccessary) with the animation starting position
+      setTimeout =>
+        # Enable transitions
+        @element.removeClass 'animation-disabled'
+
+        # Replace position classes to trigger animation
+        @$enterTile.add(@$exitTile).switchClass 'start', 'end'
+      , 10
 
     # Find possible links throughout the entire page and set meta data on them
     #
     _setupLinks: ->
       $('[data-tiler-link]').each ->
-        tileId = $(@).data('tiler-link').split(':')
+        # Get tile id (and optional tiler viewport id)
+        tileIds = $(@).data('tiler-link').split(':').reverse()
```

Comment:
just to help normalizing it. this way the tile is always at index 0 and the optional namespace is in index 1